src/bin/pg_autoctl/fsm_transition.c

```diff
@@ -471,6 +471,12 @@ fsm_rewind_or_init(Keeper *keeper)
 		}
 	}
 
+	if (!primary_drop_replication_slot(postgres, config->replication_slot_name))
+	{
+		log_warn("Failed to drop replication slot \"%s\" used by the standby failed",
+				  config->replication_slot_name);
```

Comment:
I think we should return false here, making sure that the transition fails and is retried. I can't see a good reason to continue as if everything was good when we fail to drop the replication slot. Also, the WARN should probably be raised to an ERROR.

Reply:
changed accordingly